File: Leetcode/Review/20211003/getKthFromEnd.go
package _0211003

func getKthFromEnd(head *ListNode, k int) *ListNode {
	dummyHead:=&ListNode{Next:head}
	fast:=dummyHead
	for k!=0{
		if fast!=nil{
			fast=fast.Next
			k--
		}else{
			return nil
		}
	}
	slow:=dummyHead
	for fast!=nil{
		fast=fast.Next
		slow=slow.Next
	}
	return slow
}

File: Leetcode/剑指offer/easy/20210913/getKthFromEnd.cpp
class Solution {
public:
    ListNode* getKthFromEnd(ListNode* head, int k) {
        ListNode *pre=new ListNode(-1);
        pre->next=head;
        while(k--){
            if(pre==NULL){
                return NULL;
            }
            pre=pre->next;
        }
        while(pre->next!=NULL){
            head=head->next;
            pre=pre->next;
        }
        return head;
    }
};
